### PR TITLE
IGNITE-13931 .NET: Add tests for overload service methods

### PR DESCRIPTION
**Pull request by nizhikov**
This PR is a follow-up of c2204cda29e70294cc93756eabc844b64e07a42e. It adds new tests to ensure that .Net -> .Net service call for overloaded methods works as expected.

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

**Comment by nizhikov** [comment]
Hello @ptupitsyn. Please, take a look at my changes.

This PR is a follow-up of c2204cd. It adds new tests to ensure that .Net -> .Net service call for overloaded methods works as expected.

File: modules/platforms/dotnet/Apache.Ignite.Core.Tests/Services/ServicesTest.cs [hunk 1]
```diff
@@ -1328,6 +1341,15 @@ public interface ITestIgniteService : IService, ITestIgniteServiceBase
 
             /** */
             object ErrMethod(object arg);
+
+            /** */
+            int testOverload(int count, Employee[] emps);
```
Comment (ptupitsyn) [inline modules/platforms/dotnet/Apache.Ignite.Core.Tests/Services/ServicesTest.cs:1346]: Please follow .NET naming conventions - public member names should start with a capital letter.

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: modules/platforms/dotnet/Apache.Ignite.Core.Tests/Services/ServicesTest.cs (hunk 1 above)
Comment (nizhikov) [reply modules/platforms/dotnet/Apache.Ignite.Core.Tests/Services/ServicesTest.cs]: Fixed. Thanks.

**Comment by nizhikov** [comment]
https://ci.ignite.apache.org/viewLog.html?buildId=5811357&buildTypeId=IgniteTests24Java8_RunAllNet - tests.

**Review by ptupitsyn — APPROVED** [review]

**Pull request merged**